### PR TITLE
have aria-label on feedback button indicate when there is a message

### PR DESCRIPTION
**Pull request by Alex-Jordan**
This changes the aria label on a feedback button to be like "Incorrect with message" when there is a message. This is for people who can't see the little yellow pip in the upper corner.

### CODE REVIEW

File: macros/PG.pl [hunk 1]
```diff
@@ -1207,8 +1207,9 @@ sub ENDDOCUMENT {
 						type  => 'button',
 						class => "ww-feedback-btn btn btn-sm $options{btnClass} $options{btnAddClass}"
 						. ($rh_envir->{showMessages} && $ansHash->{ans_message} ? ' with-message' : ''),
-						'aria-label' => $options{resultTitle},
-						data         => {
+						'aria-label' => $options{resultTitle}
+						. ($rh_envir->{showMessages} && $ansHash->{ans_message} ? ' with message' : ''),
```
Comment (drgrice1) [inline macros/PG.pl:1211]: This should be translated.  If you need help with that, let me know.

Comment (Alex-Jordan) [reply macros/PG.pl:1211]: Is it just ` maketext(' with message')`? Or should it encompass the `$options{resultTitle}` too?

Comment (drgrice1) [reply macros/PG.pl:1211]: It should include the title.  So this should be `maketext('[_1] with message', $options{resultTitle})` in the case that there is a message.

**Comment by drgrice1** [comment]
Although, there is something else that seems wrong with this change.  The message that this `aria-label` is being made to refer to is also added in a screen reader only span also with the result title (on lines 1192-1201).  So a screen reader user would already have heard the message that they are now being told is yet to come since that is before this in the DOM.  So should that screen reader only span be removed with this change?

**Comment by Alex-Jordan** [comment]
This is a sidebar. In testing things for this, I discovered I needed to restart `webwork2` to see the edits to `macros/PG.pl` take effect. Is that how it should be? In my head, anything in `macros/` (as opposed to `lib/`) is not part of the startup process.

**Comment by drgrice1** [comment]
Nothing in `macros/` requires a restart to `webwork2` with the exception of `macros/PG.pl`.  The safe compartment is initialized and cached at the time that the server starts, and that precompiles `macros/PG.pl`.  Both `webwork2` and the standalone renderer use this process now.  This is why when you are using `morbo` it hot reloads when there is a change to `macros/PG.pl` (but that is not done for any other macro).

**Comment by drgrice1** [comment]
I highly recommend that you try to get a local setup of webwork2 using morbo for development.  It is highly more efficient than trying to develop in the production environment with hypnotoad.

**Comment by Alex-Jordan** [comment]
The screen reader user moves past the answer blank and hears a feedback message like "Incorrect Your answer is not a reduced fraction" (or whatever). Then they move into the feedback button where they hear something like "Incorrect with message, button". Is that really a problem? They have already heard the message, and are also being told that if they activate the button they can hear it again.

**Comment by Alex-Jordan** [comment]
BTW, "Incorrect Your answer is not a reduced fraction" is actually `<span>Incorrect</span><span>Your answer is not a reduced fraction</span>`. with no space between the first tow words. The span delimiters get the screen reader to read it correctly, but in a visualization (like from VoiceOver) you see "IncorrectYour answer is not a reduced fraction".

If there is no objection, I will add a space between the two spans as part of this PR.

**Comment by drgrice1** [comment]
> The screen reader user moves past the answer blank and hears a feedback message like "Incorrect Your answer is not a reduced fraction" (or whatever). Then they move into the feedback button where they hear something like "Incorrect with message, button". Is that really a problem? They have already heard the message, and are also being told that if they activate the button they can hear it again.

That is fine I guess.  I just don't understand why they need both.

As to your question about the space ... yes, go ahead and add a space.

**Comment by Alex-Jordan** [comment]
I think it will raise issues for an accessibility tester if there is a visual signal of something on the button (the pip), but there is no corresponding signal for the AT user.

**Comment by drgrice1** [comment]
That is why I suggested removing the screen reader only span with the title and message.  It seems to me that doing so makes things more equal for those using a screen reader to the rest of us.  Why do they get the title and message before requesting it?

**Comment by Alex-Jordan** [comment]
OK, this has that space and translation on the aria.

**Comment by Alex-Jordan** [comment]
Oh I missed that suggestion. I think it would be good to remove it. If you are not looking for the feedback message, I can see it being unwanted and having it put in front of you as an intrusion. I'll remove it.

**Review by drgrice1 — APPROVED** [review]

**Review by pstaabp — APPROVED** [review]

**Pull request merged**